networkit/cpp/scd/CliqueDetect.cpp

```diff
@@ -0,0 +1,158 @@
+#include <unordered_map>
+
+#include <networkit/auxiliary/UniformRandomSelector.hpp>
+#include <networkit/clique/MaximalCliques.hpp>
+#include <networkit/graph/GraphBuilder.hpp>
+#include <networkit/scd/CliqueDetect.hpp>
+
+namespace NetworKit {
+
+CliqueDetect::CliqueDetect(const Graph &G) : SelectiveCommunityDetector(G) {
+    if (G.numberOfSelfLoops() > 0)
+        throw std::runtime_error("CliqueDetect works only with simple graphs.");
+}
+
+std::set<node> CliqueDetect::expandOneCommunity(node s) {
+    // 1. get the maximum clique in neighbors(s) + s
+    std::set<node> result;
+    result.insert(s);
+
+    std::vector<node> sn;
+    sn.reserve(G->degree(s));
+    G->forNeighborsOf(s, [&](node v) { sn.push_back(v); });
+
+    if (!sn.empty()) {
+        for (node u : getMaximumWeightClique(sn, [s](node v) { return v == s; })) {
+            result.insert(sn[u]);
+        }
+    }
+
+    return result;
+}
+
+std::set<node> CliqueDetect::expandOneCommunity(const std::set<node> &seeds) {
+    std::set<node> result(seeds);
+
+    if (!seeds.empty()) {
+        std::unordered_map<node, count> sn;
+        G->forNeighborsOf(*seeds.begin(), [&](node v) {
+            if (seeds.find(v) == seeds.end()) {
+                sn.insert({v, 1u});
+            }
+        });
+
+        for (auto it = ++seeds.begin(); it != seeds.end(); ++it) {
+            G->forNeighborsOf(*it, [&](node v) {
+                auto it = sn.find(v);
+                if (it != sn.end()) {
+                    ++it->second;
+                }
+            });
+        }
+
+        std::vector<node> neighbors;
+        neighbors.reserve(sn.size());
+        for (auto it : sn) {
+            assert(it.second <= seeds.size() && "Graph has probably multi-edges!");
+            if (it.second == seeds.size()) {
+                neighbors.push_back(it.first);
+            }
+        }
+
+        if (!neighbors.empty()) {
+            const std::vector<node> clique = getMaximumWeightClique(
+                neighbors, [&](node v) { return seeds.find(v) != seeds.end(); });
+
+            for (node u : clique) {
+                result.insert(neighbors[u]);
+            }
+        }
+    }
+
+    return result;
+}
+
+template <typename F>
+std::vector<node> CliqueDetect::getMaximumWeightClique(const std::vector<node> &nodes,
+                                                       F is_seed) const {
```

Comment:
Use lowerCamelCase for `is_seed`

Reply:
Fixed in 35b4f77ef9b92eb98e5c2c0e19ccfe6a7aa3620e